transport-native-unix-common/src/main/c/netty_unix_socket.c

```diff
@@ -319,11 +324,13 @@ static jint _sendTo(JNIEnv* env, jint fd, jboolean ipv6, void* buffer, jint pos,
     ssize_t res;
     int err;
     do {
-       res = sendto(fd, buffer + pos, (size_t) (limit - pos), 0, (struct sockaddr*) &addr, addrSize);
+       res = sendto(fd, buffer + pos, (size_t) (limit - pos), fastOpen == JNI_TRUE ? MSG_FASTOPEN : 0, (struct sockaddr*) &addr, addrSize);
        // keep on writing if it was interrupted
     } while (res == -1 && ((err = errno) == EINTR));
 
-    if (res < 0) {
+    if (res == -1 && err == EINPROGRESS && fastOpen == JNI_TRUE) {
+        return 0;
```

Comment:
I think it would be "cleaner" to let `EINPROGRESS` bubble up to the java layer and handle it explicit there.  This will also make the error handling more robust.  (related to my comment on connect0(...)